src/masternodes/mn_checks.cpp

```diff
@@ -2385,7 +2400,8 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         DCT_ID BTC = FindTokenByPartialSymbolName(CICXOrder::TOKEN_BTC);
         if (order->idToken == BTC && order->orderPrice == COIN) {
```

Comment:
Let's abstract a bit more so it's clearer, self documenting on what it does and take the bugged code out of the main path. 

``
```
// This is a function for bugged blockchain region check, and maintained purely for maintaining compatibility with older network data.
ICXLegacyBugPath(params..) {
 lambda IsICXBuggedRegion(network, height) {
   
  }
  
  lamda BuggedICXTransfer() { .. }

  IsICXBuggedRegion(network, height) {
   ..BuggedICXTransfer();
  }
}



... 

if (!ICXLegacyBugPath(params)) {

}
```